modules/EPMC.py

```diff
@@ -0,0 +1,179 @@
+    logging.info(f'Output file: {outputFile}')
+    logging.info(f'Generating evidence:')
+
+    ##
+    ## Load/filter datasets:
+    ##
+    filtered_cooccurrence_df = (
+        # Reading file:
+        spark.read.parquet(cooccurrenceFile)
+
+        # Filtering for disease/target cooccurrences:
+        .filter(
+            (pf.col('type') == 'GP-DS') &  # Filter gene/protein - disease cooccurrence
+            (pf.col('isMapped') == True) &  # Filtering for mapped cooccurrences
+            (pf.col('pmid').isNotNull()) &  # Excluding publications without pmid
+            (pf.length(pf.col('text')) < 600) & # Exclude sentences with more than 600 characters
+            (pf.col('label1').isin(EXCLUDED_TARGET_TERMS) == False) # Excluding target labels from the exclusion list
+        )
+        # Renaming columns:
+        .withColumnRenamed('keywordId1', 'targetFromSourceId')
+        .withColumnRenamed('keywordId2', 'diseaseFromSourceMappedId')
+        .withColumnRenamed('label1', 'targetFromSource')
+        .withColumnRenamed('label2', 'diseaseFromSource')
+
+        .withColumn('tmp', pf.col('pmid').cast(StringType()))
+    )
+
+    # Report on the number of diseases, targets and associations if loglevel == "debug" to avoid cost on computation time:
+    logging.debug(f"Number of publications: {filtered_cooccurrence_df.select(pf.col('tmp')).distinct().count()}")
+    logging.debug(f"Number of targets: {filtered_cooccurrence_df.select(pf.col('targetFromSourceId')).distinct().count()}")
+    logging.debug(f"Number of diseases: {filtered_cooccurrence_df.select(pf.col('diseaseFromSourceMappedId')).distinct().count()}")
+    logging.debug(f"Number of associations: {filtered_cooccurrence_df.select(pf.col('diseaseFromSourceMappedId'), pf.col('targetFromSourceId')).dropDuplicates().count()}")
+
+    # Aggregating cooccurrence, get score apply filter:    
+    aggregated_df = (
+        filtered_cooccurrence_df
+
+        # Aggregating data by publication, target and disease:
+        .groupBy(['pmid', 'targetFromSourceId', 'diseaseFromSourceMappedId'])
+        .agg(
+            pf.first(pf.col('targetFromSource')).alias('targetFromSource'),
+            pf.first(pf.col('diseaseFromSource')).alias('diseaseFromSource'),
+            pf.collect_set(pf.col('tmp')).alias('literature'),
+            pf.collect_set(
+                pf.struct(  
+                    pf.col('text'),
+                    pf.col('start1').alias('tStart'),
+                    pf.col('end1').alias('tEnd'),
+                    pf.col('start2').alias('dStart'),
+                    pf.col('end2').alias('dEnd'),
+                    pf.col('section')
+                )
+            ).alias('textMiningSentences'),
+            pf.sum(pf.col('evidence_score')).alias('resourceScore')
+        )
+
+        # Summarizing all scores then filter for evidence string with at least score == 2
+        .filter(pf.col('resourceScore') > 1)
```

Comment:
The comment implies filtering on `>= 2`, but the actual filter is `> 1`. Granted, if those scores are always integers, then it's the same thing, but I think it's still better if the semantics is the same in the code and in the comment.

Also, I'm a bit confused, isn't the score supposed to be normalised to [0, 1] already? Or is this something done by the ETL now?

Reply:
The raw scores for a single cooccurrence range between 1-10, these scores for all cooccurrences then summarized for one publication to get the `associationScore`. This is what the parser generates for the ETL. 

This score is then divided by 100 and capped at 1. All within ETL.